client/state/data-layer/wpcom/read/tags/index.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	READER_TAGS_REQUEST,
+} from 'state/action-types';
+import {
+	receiveTags,
+} from 'state/reader/tags/items/actions';
+
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export function requestTags( store, action, next ) {
+	const path = action.payload && action.payload.slug
+		? `/read/tags/${ action.payload.slug }`
+		: '/read/tags';
+
+	store.dispatch( http( {
+		path,
+		method: 'GET',
+		apiVersion: '1.2',
+		onSuccess: action,
+		onFailure: action,
+	} ) );
+
+	next( action );
+}
+
+/**
+ * Normalize response from the api so whether we get back a single tag or a list of tags
+ * we always pass forward a list
+ * @param  {Tag|Tags} apiResponse api response from the tags endpoint
+ * @return {Tags}             An object containing list of tags
+ */
+function fromApi( apiResponse ) {
+	let tags;
+	if ( apiResponse.tag )	 {
+		tags = [ apiResponse.tag ];
+	} else if ( apiResponse.tags ) {
+		tags = map( apiResponse.tags, tag => ( { ...tag, is_following: true } ) );
+	} else {
+		if ( process.env.NODE_ENV === 'development' ) {
+			throw new Error( 'bad api response for /read/tags' );
+		}
+		tags = [];
+	}
+
+	return { tags };
+}
+
+export function receiveTagsSuccess( store, action, next, apiResponse ) {
+	store.dispatch( receiveTags( {
+		payload: fromApi( apiResponse ),
```

Comment:
I realize we're working with existing code here which is sad since it's designed to be tightly-coupled into the specifics of the PHP API. It would be awesome if we didn't have a `RECEIVE_TAGS_SUCCESS` action taking a `payload` and instead could use something like `TAGS_ADD` with specific properties related to _tags_ instead of _APIs_